lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,1119 @@
+-- Equivalent to GLua Entity:Getcontents.
+-- You may prefer to use Entity:getShipmentContents instead, although that function is slightly slower.
+-- @return number? Index of contents, or nil if not a shipment
+function ents_methods:getShipmentContentsIndex()
+	self = getent(self)
+	return self.Getcontents and assertsafety(self:Getcontents()) or nil
+end
+
+--- Get the info for the contents of the shipment. DarkRP only.
+-- Equivalent to "darkrp.getCustomShipments()[ent:getShipmentContentsIndex()]"
+-- @return table? Contents, or nil if not a shipment
+function ents_methods:getShipmentContents()
+	self = getent(self)
+	if not CustomShipments or not self.Getcontents then return end
+	return instance.Sanitize(CustomShipments[self:Getcontents()])
+end
+
+-- Player methods
+
+if SERVER then
+	--- Unown every door and vehicle owned by this player. DarkRP only.
+	-- @server
+	function player_methods:keysUnOwnAll()
+		self = getply(self)
+		if instance.player ~= SF.Superuser and instance.player ~= self then SF.Throw("may not use this function on anyone other than owner", 2) return end
+		assertsafety(self:keysUnOwnAll())
+	end
+	player_methods.keysUnownAll = player_methods.keysUnOwnAll
+	
+	--- Returns the time left on a player's team ban. DarkRP only.
+	-- Only works if the player is the owner of the chip, or if the chip is running in superuser mode.
+	-- @server
+	-- @param number? team The number of the job (e.g. TEAM_MEDIC). Uses the player's team if nil.
+	-- @return number? The time left on the team ban in seconds, or nil if not banned.
+	function player_methods:teamBanTimeLeft(team)
+		if team ~= nil then checkluatype(team, TYPE_NUMBER) end
+		self = getply(self)
+		if instance.player ~= SF.Superuser and instance.player ~= self then SF.Throw("may not use this function on anyone other than owner", 2) return end
+		return assertsafety(self:teamBanTimeLeft())
+	end
+	
+	--- Request money from a player.
+	-- This is subject to a burst limit. Use the darkrp.canMakeMoneyRequest function to check if you can request money that tick.
+	-- @server
+	-- @param string? message An optional custom message that will be shown in the money request prompt. May not exceed 60 bytes in length.
+	-- @param number amount The amount of money to ask for.
+	-- @param function? callbackSuccess Optional function to call if request succeeds.
+	-- @param function? callbackFailure Optional function to call if request fails.
+	-- @param Player? receiver The player who may or may not receive the money, or the owner of the chip if not specified. Superuser only.
+	function player_methods:requestMoney(message, amount, callbackSuccess, callbackFailure, receiver)
+		-- Argument order is different for purposes of compatibility with https://github.com/loganlearner/starfall-darkrp-library
+		return darkrp_library.requestMoney(self, amount, message, callbackSuccess, callbackFailure, receiver)
```

Comment:
IMO, Starfall should not be making any assumptions about the chip's environment. It should be totally fine for a chip to set `darkrp.requestMoney` to `nil` and continue to use `Player:requestMoney`.

Reply:
Yeah, that's true